magicmaze.js

```diff
@@ -775,6 +811,8 @@ function (dojo, declare) {
       this.explores = new Map()
       this.escalators = new Map()
       this.escalatorEls = new Map()
+      this.escalatorEnds = new Map() // bidirectional mapping instead of
```

Comment:
minor: consider adding a comment on escalators now that this comment probably prompts the question "which element is in escalators??", something like `// Canonical element is the one with the lower x` which was placed elsewhere